keps/sig-storage/695-skip-permission-change/README.md

```diff
@@ -0,0 +1,276 @@
+    // and emptydir.
+    // + optional
+    FSGroupChangePolicy *PodFSGroupChangePolicy
+}
+```
+
+### Risks and Mitigations
+
+- One of the risks is if user volume's permission was previously changed using old algorithm(which changes permission of top level directory first) and user opts in for `OnRootMismatch` `FSGroupChangePolicy` then we can't distinguish if the volume was previously only partially recursively chown'd.
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate
+    - Feature gate name: ConfigurableFSGroupPolicy
+    - Components depending on the feature gate: kubelet, kube-apiserver, kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  No enabling the feature gate does not change any default behaviour.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+  Yes the feature-gate can be disabled once enabled. This will cause all volume mounts that
+  require permission change to use recursive permission and ownership change. This is the default
+  behaviour without the feature gate.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  For pods that have expected value in `pod.Spec.SecurityContext.FSGroupChangePolicy` as defined in this KEP,
+  it will start using specified policy.
+
+* **Are there any tests for feature enablement/disablement?**
+  There aren't any e2e but there are unit tests that cover this behaviour.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Since this feature requires users to opt-in by setting new field in pod spec, it should
+  not impact already running workloads.
+
+* **What specific metrics should inform a rollback?**
+  If after enabling this feature and setting `pod.Spec.SecurityContext.FSGroupChangePolicy`
+  users notice an increase in volume mount time via `storage_operation_duration_seconds{operation_name=volume_mount}`
+  then a cluster-admin may want to rollback the feature.
+
+* **Were upgrade and rollback tested? Was upgrade->downgrade->upgrade path tested?**
+  We have not fully tested upgrade and rollback. We have unit tests that cover the scenario
+  of feature gate being enabled and then disabled. But we will need to do more upgrade->downgrade->upgrade testing.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features,
+  APIs, fields of API types, flags, etc.?**
+  This feature deprecates no existing functionality.
+
+
+### Monitoring requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  Not applicable.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to
+  determine the health of the service?**
+  - [ ] Metrics
```

Comment:
Can we not use the volume mount metric? Additionally, could we have a new metric that breaks out the ChangeVolumeOwnership latency from MountVolume?

Reply:
We could but the next section asks for hard numbers if I am reading this right:

> What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**

How do we fill that? 95% of mounts should succeed? 99 percentile of mount timing should be within 5s? where does these numbers come from? 

May be I could just fill SLIs but not fill section about SLOs...

Reply:
cc @johnbelamaric

Reply:
I have filled out metrics for SLIs. However, I am hesitant to put hard number of SLOs and hence have suggested admin to watch out for an increase in mount errors or mount timings (no hard percentage based numbers).